Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
```

Comment:
I don't know that there's an established naming convention for things that return `Result<Self>`, but I'm open to suggestions.

Reply:
The stdlib seems to use [`try_new`](https://github.com/rust-lang/rust/blob/master/src/librustc_data_structures/owning_ref/mod.rs#L889) or [`from_*`](http://doc.rust-lang.org/1.32.0/alloc/str/trait.FromStr.html#tymethod.from_str) for these.

Reply:
Is it idiomatic to only return `Self` from `new`? I haven't found any examples showing otherwise. I lean towards *not* returning `Result<Self>` and just returning `Self` if that be the case. It seems like this particular lint is tied closely with `new_without_default`.

Reply:
> Is it idiomatic to only return `Self` from `new`?

Can you find an example in the stdlib that returns something other than `Self` for `new`?

Reply:
I can't, though that's just from me clicking through. I don't how to search specifically for that.

Reply:
Hrmm... `try_new` might be interesting to try on for size. I think I'd like to boil off the other lints before tackling that, though, so I'll be 👎 on this for now.

Reply:
> @baumanj are you saying that we _should_ be returning `Self` from `new()` and _not_ `Result<Self>`? If you are, then I think we are agreeing with each other.

The lint [`clippy::new_ret_no_self`](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#new_ret_no_self) asserts that any associated function called `new` should be returning `Self`. I agree that is the rust convention, but I also think that `Result<Self>` is an acceptable variant since it's sill a *kind* of `Self`. Given the static typing, it's not like you're going to forget to handle the error case, so I don't know if there's a ton of value in `try_new`, so I'm suggesting we keep this in `ALLOWED_LINTS`. That said, I don't feel terribly strongly either way.

Reply:
I don't have a strong opinion on this one.

Reply:
Where the code is now I'd agree with @baumanj and say this is ok, in that it's a kind of `Self`.  

One pattern I've noticed is where we build up a struct in a `new`, by calling other `new` -> `Result<Self>` functions which ends up spreading out through the code. There seems to be [cases](https://github.com/habitat-sh/habitat/blob/master/components/sup/src/manager/service/mod.rs#L187-L188) where all we're doing is creating directories or some FS ops which can return e.g. permission denied.  There could be cases where we do this later in the flow on-demand rather than up front.